### PR TITLE
Added docstrings to AIPlugIn

### PR DESCRIPTION
**Pull request by asgibson**
Here is an update for ai_plugin.py using Google style docstrings. Also using type annotations.

Is this the style we'd like to go forward with?
There are other style choices we could choose. i.e., NumPy, reST, or Epytext (others?)

Opinions?

### CODE REVIEW

File: onair/src/ai_components/ai_plugin_abstract/ai_plugin.py [hunk 1]
```diff
@@ -1,38 +1,63 @@
...
-"""
+
+
 class AIPlugIn(ABC):
-    def __init__(self, _name, _headers):
-        """
-        Superclass for data driven components: VAE, PPO, etc. Allows for easier modularity.
+    """This serves as a base for all plugins.
+
+    This is the superclass for data driven components: VAE, PPO, etc. It is
```
Comment (the-other-james) [inline onair/src/ai_components/ai_plugin_abstract/ai_plugin.py:17]: We should probably update the examples to ones we have... which is none (the Kalman plugin doesn't count, right?)... so I don't know.

We could at least get rid of the acronyms.

**Comment by the-other-james** [comment]
I don't have a strong preference for the comment style we use. If I had to pick, I would go with whatever the python libraries use.

File: onair/src/ai_components/ai_plugin_abstract/ai_plugin.py (hunk 1 above)
Comment (asgibson) [reply onair/src/ai_components/ai_plugin_abstract/ai_plugin.py:17]: I'd be in favor of just removing that sentence for now.

**Comment by asgibson** [comment]
We have decided to follow numpydoc style:
https://numpydoc.readthedocs.io/en/latest/format.html

**Comment by dragonejt** [comment]
Hello, I've been browsing the other PR's and issues in the repository. I just wanted to note that since you all have decided on a docstring style to move forward with, most IDEs now have support for automatically generating docstring formats/templates for you to fill in. Essentially, all you will have to do is type `"""` and press enter, and a docstring template will be generated based on your function parameters and return values. This will save you a lot of time vs typing out the docstrings in full by yourself.

In VSCode, you can use the [autoDocstring](https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring) extension, which has support for numpydoc.

PyCharm already has [automatic docstring templates](https://www.jetbrains.com/help/pycharm/settings-tools-python-integrated-tools.html) builtin, and already supports numpydoc.

**Comment by asgibson** [comment]
@dragonejt, thank you for the tool suggestions. As long as people follow the NumPy docstring formatting, they can use whatever tool they want to create them.

File: onair/src/ai_components/ai_plugin_abstract/ai_plugin.py (hunk 1 above)
Comment (asgibson) [reply onair/src/ai_components/ai_plugin_abstract/ai_plugin.py]: Done.

File: onair/src/ai_components/ai_plugin_abstract/ai_plugin.py [hunk 2]
```diff
@@ -1,40 +1,107 @@
...
+    meant to induce standards and structures of compliance for imported
+    plugins: Knowledge Representation, Learner, Planner or Complex Reasoner.
+    Those plugins must inherit this class.
+
+    Attributes
+    ----------
+    component_name : str
+        Name given to instance of plugin.
+    headers : list of str
+        Names for each data point in the OnAIR data frame.
```
Comment (the-other-james) [inline onair/src/ai_components/ai_plugin_abstract/ai_plugin.py:41]: I think you should specify that the headers are for the low_level_data list in the update function.

**Review by the-other-james — CHANGES_REQUESTED** [review]
Looks good, thank you for getting us on track for having a set style!

Just one small suggestion, otherwise good to merge.

**Review by the-other-james — APPROVED** [review]

**Pull request merged**